### PR TITLE
ISSUE-4 User auto provisioning

### PR DESCRIPTION
**Pull request by chibenwa**
Use of the play client livrairy to pre-configure the James server.
Random domain name and username generation is based on UUIDs.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -31,6 +31,4 @@ FQDN :
 org.apache.james.gatling.jmap.scenari.JmapAuthenticationScenario
 ----
 
-Authenticate a single JMAP user, one time.
-
-You need to execute the provisioning script with as a first parameter the docker ID of the James container before running it.
\ No newline at end of file
+Authenticate 10 JMAP user, one time. Demonstrate user auto-provisioning threw James WebAdmin server.
```
Comment (mbaechler) [inline README.adoc]: through

File: src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministration.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.apache.james.gatling.control
+
+import play.api.libs.ws.ning.NingWSClient
+
+object JamesWebAdministration {
+
+  val wsClient = NingWSClient()
+
+  val baseUrl: String = "http://172.17.0.4:8000"
```
Comment (mbaechler) [inline src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministration.scala]: you should retrieve the target server through an env variable (and fallback to localhost)

Comment (mbaechler) [inline src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministration.scala]: don't you have a type for URL ?

File: src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministration.scala [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  def addDomain(domain: String) = wsClient.url(s"$baseUrl/domains/$domain")
```
Comment (mbaechler) [inline src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministration.scala]: in scala, you have value types, please use them (and define a Domain object, a User object, etc), it's so cheap to do that you don't have any excuse not to use them

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala [hunk 4]
```diff
@@ -1,18 +1,38 @@
...
 class JmapAuthenticationScenario extends Simulation {
 
-  val domain = "domain-jmapauthentication.tld"
-  val username = "username@" + domain
-  val password = "password"
+  val userCount = 100
+
+  val domain = UUID.randomUUID().toString
+
+  JamesWebAdministration.addDomain(domain).get
```
Comment (mbaechler) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala]: what is the get for ?

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala [hunk 5]
```diff
@@ -1,18 +1,38 @@
...
+
+  val loginsAndPassWords = Range.apply(0, userCount)
```
Comment (mbaechler) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala]: `(0 until userCount)`

File: src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministration.scala (hunk 2 above)
Comment (chibenwa) [inline src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministration.scala]: https://github.com/linagora/james-gatling/issues/9 is what you describe.

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala (hunk 4 above)
Comment (chibenwa) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala]: This get is to await future for domain creation and avaid concurency issue.

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala (hunk 5 above)
Comment (chibenwa) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala]: ok

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala (hunk 4 above)
Comment (mbaechler) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapAuthenticationScenario.scala]: ok, so this time you could put a return type on addDomain (:

**Comment by mbaechler** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

File: README.adoc [hunk 6]
```diff
@@ -31,6 +31,4 @@ FQDN :
 org.apache.james.gatling.jmap.scenari.JmapAuthenticationScenario
 ----
 
-Authenticate a single JMAP user, one time.
-
-You need to execute the provisioning script with as a first parameter the docker ID of the James container before running it.
\ No newline at end of file
+Authenticate 10 JMAP user, one time. Demonstrate user auto-provisioning through James WebAdmin server.
```
Comment (rouazana) [inline README.adoc]: users

**Pull request merged**